ruma-events-macros/src/collection.rs

```diff
@@ -139,17 +153,18 @@ fn to_event_content_path(
 
     assert_eq!(&name[..2], "m.");
 
-    let event_str = name[2..].split('.').last().unwrap();
+    let path = name[2..].split('.').collect::<Vec<_>>();
 
+    let event_str = path.last().unwrap();
     let event = event_str
         .split('_')
         .map(|s| s.chars().next().unwrap().to_uppercase().to_string() + &s[1..])
         .collect::<String>();
 
-    let module = Ident::new(event_str, span);
     let content_str = Ident::new(&format!("{}EventContent", event), span);
+    let path = path.iter().map(|s| Ident::new(s, span));
     syn::parse_quote! {
-        ::ruma_events::room::#module::#content_str
+        ::ruma_events::#( #path::)*#content_str
```

Comment:
I *think* this should work too (and is easier to read IMO):

```suggestion
        ::ruma_events::#(#path)::*::#content_str
```